### PR TITLE
Added all files to the new repository

### PR DESCRIPTION
**Pull request by ZaMaZaN4iK**
Hello. I added manually files to the this repository. Check it, if you want. Because i think you are hard working now, and manually inserting of my algorithm is my work :-) 

Sorry, that i didn't know, that this repository is main. My bad.

### CODE REVIEW

File: include/boost/algorithm/is_palindrome.hpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+/// \file  is_palindrome.hpp
+/// \brief Checks the input sequence on palindrome.
+/// \author Alexander Zaitsev
+
+#ifndef BOOST_ALGORITHM_IS_PALINDROME_HPP
+#define BOOST_ALGORITHM_IS_PALINDROME_HPP
+
+#include <iterator>
+
```
Comment (mclow) [inline include/boost/algorithm/is_palindrome.hpp]: You should `#include <functional>` here, to get the definition of `std::is_equal`

File: test/is_palindrome_test.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        return v1 == v2;
+    }
+};
+
+
+void test_is_palindrome()
+{
+    const std::list<int> empty;
+    const std::vector<char> singleElement{'z'};
```
Comment (mclow) [inline test/is_palindrome_test.cpp]: If you write this as `const std::vector<char> singleElement(1, 'z');` then it works on C++03 as well.

**Comment by mclow** [comment]
Other than those two nits, this looks fine.  Fix them, and I'll merge it.

File: include/boost/algorithm/is_palindrome.hpp (hunk 1 above)
Comment (ZaMaZaN4iK) [inline include/boost/algorithm/is_palindrome.hpp]: Fixed

**Pull request merged**